src/drive/web/modules/actions/index.jsx

```diff
@@ -350,19 +335,25 @@ export const destroy = ({ pushModal, popModal }) => {
     name: 'destroy',
     icon: 'trash',
     action: files =>
-      pushModal(<DestroyConfirm files={files} onClose={popModal} />),
-    Component: function Destroy(props) {
+      pushModal(
+        <DestroyConfirm
+          files={Array.isArray(files) ? files : [files]}
+          onClose={popModal}
+        />
+      ),
+    Component: forwardRef(function Destroy(props, ref) {
       const { t } = useI18n()
       return (
-        <ActionMenuItem
-          onClick={() =>
-            pushModal(<DestroyConfirm files={props.files} onClose={popModal} />)
-          }
-          left={<Icon icon={TrashIcon} color="var(--errorColor)" />}
-        >
-          <span className="u-error">{t('SelectionBar.destroy')}</span>
-        </ActionMenuItem>
+        <ActionsMenuItem {...props} ref={ref}>
+          <ListItemIcon>
+            <Icon icon={TrashIcon} color="var(--errorColor)" />
+          </ListItemIcon>
+          <ListItemText
+            className="u-error"
+            primary={t('SelectionBar.destroy')}
+          />
+        </ActionsMenuItem>
       )
-    }
+    })
   }
 }
```

Comment:
I didn't understand why you have to check isArray() everywhere and then [files]. Can you explain?

Reply:
Your comment makes me think that I haven't documented this point. 

Actions are used in two places in a file's drop-down menu and in the selection bar. Before the migration to ActionsMenu, we always received an array in both case. The ActionsMenu now sends the document directly to the action, while the SelectionBar sends an array of documents. This code seeks to manage the two cases.

I'm going to add a comment in the commit for the future me